storage/src/main/java/tech/pegasys/teku/storage/server/state/FinalizedStateCache.java

```diff
@@ -106,13 +136,21 @@ private BeaconState regenerateState(final UInt64 slot, final BeaconState stateFr
                       stateFromCache.getSlot().compareTo(stateFromDisk.getSlot()) >= 0)
               .orElse(stateFromDisk);
       if (preState.getSlot().equals(slot)) {
-        return preState;
+        return SafeFuture.completedFuture(preState);
+      }
+      final long regenerateSlotCount = slot.minusMinZero(stateFromDisk.getSlot()).longValue();
+      LOG.trace("Slots to regenerate state from: {}", regenerateSlotCount);
+      if (regenerateSlotCount > 10_000L) {
```

Comment:
we could have this constant statically defined up in the class.

Reply:
true, potentially at some point configurable, but initially 10k is a decent distance